packages/atlas/src/components/Text/Text.styles.ts

```diff
@@ -3,9 +3,37 @@ import styled from '@emotion/styled'
 
 import { cVar, sizes } from '@/styles'
 
+type ColorVariant =
+  | 'default'
+  | 'muted'
+  | 'strong'
+  | 'primary'
+  | 'error'
+  | 'success'
+  | 'caution'
+  | 'neutral100'
+  | 'neutral200'
+  | 'neutral200Lighten'
+  | 'neutral500'
+  | 'inherit'
+
+const mappedColors = {
+  default: cVar('colorText'),
+  muted: cVar('colorTextMuted'),
+  strong: cVar('colorTextStrong'),
+  primary: cVar('colorTextPrimary'),
+  error: cVar('colorTextError'),
+  success: cVar('colorTextSuccess'),
+  caution: cVar('colorTextCaution'),
+  neutral100: cVar('colorCoreNeutral100'),
+  neutral200: cVar('colorCoreNeutral200'),
+  neutral200Lighten: cVar('colorCoreNeutral200Lighten'),
+  neutral500: cVar('colorCoreNeutral500'),
```

Comment:
I don't like the fact that we have a lot of random colors here.
What will happen if a designer adds text with the color that's not on this list of mapped colors? We will need to add the next color to this mappedColors, right? 
I think this should be either:
1. only text colors(`colorTextPrimary`, `colorTextSuccess`, `colorTextMuted` etc.) 
2. all the colors we have.

If we decide to pick the first option, we will need to style all Text components with emotion where we use non-standard colors
If we decide to pick the second option we would need to extract somehow all the colors from variables.

Reply:
What do you guys think about it @kdembler @eldiegod ?

Reply:
@drillprop Maybe we should add some prop with custom color?

Reply:
Maybe just all the colors from variables then?

Reply:
Prop with custom color? How it would work? 
I'd opt for getting all the colors from tokens. The thing is that it could be problematic with filtering types from `theme` only for colors. This is what `theme` object looks like in `variables.tsx`

```tsx
const theme = {
// color variables here
// animation variables here
// effects variables here
}
```
We could create separate variables for different types of tokens. For example: 
```tsx
const colorVariables = {
...
} 
const effectVariables = {
...
}
// and then merge them
const theme = {
...colorVariables,
...effectVariables
}
```
 If you want - I could refactor the style-dictionary config to make this easier. I did some experimenting yesterday and I'm close to accomplishing that.

Reply:
We can do it with just typescript: https://www.typescriptlang.org/play?#code/MYewdgzgLgBFAWBTAtomBeGBvAUDGA5KADYgBOAGgQFyEEA0ehJ5AmjXY-gRAJYBeiDgQZMeIVAl5gA5gFFiEIbRE4AvjhxQAngAc0AMV7EoiMgGUoAQzJRpMgOq8EAHgDSibRHowACmUQAM14ADxhEENMwABMIGGgyewA+DBgPL3DIxBi4gAMAEix-INC1QulAsxgAfQo1XJgAfjTPONowRAA3My09NABhEFIydLjMIxMzSxs7WSdXAGtPEEC4PpW4JFQfIiHyAiScIA